File: backend/SindhDormitory.Domain/Entities/Application.cs
// =============================================================================
// Domain/Entities/Application.cs
// =============================================================================
using SindhDormitory.Domain.Enums;

namespace SindhDormitory.Domain.Entities;

public class Application : BaseEntity
{
    public int               ApplicationId  { get; set; }
    public int               StudentId      { get; set; }
    public int               AcademicYearId { get; set; }
    public ApplicationStatus Status         { get; set; } = ApplicationStatus.Draft;
    public DateTime?         SubmittedAt    { get; set; }
    public string?           Remarks        { get; set; }

    // Navigation
    public Student      Student      { get; set; } = null!;
    public AcademicYear AcademicYear { get; set; } = null!;
    public ICollection<ApplicationHostelPreference> Preferences    { get; set; } = [];
    public ICollection<ApplicationStatusHistory>    StatusHistory  { get; set; } = [];
    public MeritResult?  MeritResult  { get; set; }
    public ProcessingFee? ProcessingFee { get; set; }
    public ICollection<Allocation>   Allocations { get; set; } = [];
}

File: backend-node/src/entities/Application.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany, JoinColumn, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { Student } from './Student';
import { AcademicYear } from './AcademicYear';
import { ApplicationHostelPreference } from './ApplicationHostelPreference';
import { Allocation } from './Allocation';

@Entity('Applications')
export class Application {
  @PrimaryGeneratedColumn({ name: 'ApplicationId' })
  applicationId!: number;

  @Column({ name: 'StudentId', type: 'int' })
  studentId!: number;

  @Column({ name: 'AcademicYearId', type: 'int' })
  academicYearId!: number;

  @Column({ name: 'Status', type: 'varchar', length: 20, default: 'Submitted' })
  status!: string;

  @Column({ name: 'SubmittedAt', type: 'datetime', nullable: true })
  submittedAt!: Date | null;

  @CreateDateColumn({ name: 'CreatedAt' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'UpdatedAt' })
  updatedAt!: Date;

  @ManyToOne(() => Student, (s) => s.applications)
  @JoinColumn({ name: 'StudentId' })
  student!: Student;

  @ManyToOne(() => AcademicYear)
  @JoinColumn({ name: 'AcademicYearId' })
  academicYear!: AcademicYear;

  @OneToMany(() => ApplicationHostelPreference, (p) => p.application)
  preferences!: ApplicationHostelPreference[];

  @OneToMany(() => Allocation, (a) => a.application)
  allocations!: Allocation[];
}
